src/lib.rs

```diff
@@ -186,6 +247,17 @@ pub trait ARGlasses: Send {
     fn serial(&mut self) -> Result<String>;
     /// Get a single sensor event. Blocks.
     fn read_event(&mut self) -> Result<GlassesEvent>;
+    /// read until next valid event. Blocks.
+    fn next_event(&mut self) -> GlassesEvent {
```

Comment:
This is not really useful on the public interface. Instead this should be folded into the Fusion code.